packages/hawtio/src/plugins/shared/jolokia-service.ts

```diff
@@ -518,57 +523,73 @@ class JolokiaService implements IJolokiaService {
 }
 
 type JQueryBeforeSend = (this: unknown, jqXHR: JQueryXHR, settings: unknown) => false | void
+type JQueryAjaxError = (xhr: JQueryXHR, text: string, error: string) => void
 
-export type AttributeValues = { [name: string]: unknown }
+export type AttributeValues = Record<string, unknown>
 
+/* eslint-disable @typescript-eslint/no-unused-vars */
 /**
  * Dummy Jolokia implementation that does nothing.
  */
-/* eslint-disable @typescript-eslint/no-unused-vars */
-class DummyJolokia implements IJolokia {
+class DummyJolokia implements Jolokia {
```

Comment:
I have had to copy this into hawtio-online in order to always return an implementation of jolokia, regardless of error. Can we export it, please?

Of course, it might be that a review of `ManagedPod` might render this mute.

https://github.com/hawtio/hawtio-online/blob/2.x/packages/management-api/src/managed-pod.ts#L21
https://github.com/hawtio/hawtio-online/blob/2.x/packages/management-api/src/managed-pod.ts#L110

Reply:
@phantomjinx It's an important point but I don't think is related to this pull req. It's a bigger topic than briefly duscussing here, so please file a new issue to discuss it further.

Here are my points:
* As we discussed, hawtio-online should rather use `jolokiaService` instead of directly depending on `jolokia.js`. This way, I don't think we need to export this `DummyJolokia` class.
* What's important to discuss here is really that currently `jolokiaService` is provided as a singleton service, whereas hawtio-online should expect to use multiple Jolokia instances with each distinct connection, because hawtio-online needs to switch to different Jolokia endpoints provided by pods. We'd rather need to consider how to change the `jolokiaService` so that it can serve hawtio-online as well.

Let's discuss it further at a new issue.

Reply:
@phantomjinx Went ahead and filed #573 for you.